### PR TITLE
Add block/net device benchmark tests.

### PR DESCRIPTION
**Pull request by sandreim**
Signed-off-by: Andrei Sandu <sandreim@amazon.com>

## Reason for This PR

Benchmark test for net and block devices: #1097 .

## Description of Changes

- devtool: add -r option to map a ramdisk as /srv
- add env pytest marker to skip/run benchmark tests
- iperf3 based benchmark for net device
- fio based benchmark for block device
- add initial benchmarking docs

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria. Where there are two options, keep one.]`  
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] This PR is linked to an issue and the reason for this PR is
      clearly provided. 
- [x] The description of changes is clear and encompassing.
- [x] The required doc changes are included in this PR. Docs in scope are all `*.md` files
      located either in the repository root, or in the `docs/` directory.
- [x] Code-level documentation for touched code is included in this PR.
- [x] No API changes are included in this PR.
- [x] The changes in this PR have no user impact.
- [x] No new `unsafe` code has been added.

### CODE REVIEW

File: docs/benchmark.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        "read_bw_min": 229.88,
+        "read_bw_mean": 247.6,
+        "read_bw_max": 262.27,
+        "write_iops_min": 0,
+        "write_iops_mean": 0,
+        "write_iops_max": 0,
+        "write_bw_min": 0.0,
+        "write_bw_mean": 0.0,
+        "write_bw_max": 0.0
+    },
```
Comment (aghecenco) [inline docs/benchmark.md]:
```suggestion
    }
]
```

File: tests/conftest.py [hunk 2]
```diff
@@ -116,7 +116,7 @@ def test_with_any_microvm(test_microvm_any):
 session root and temporary directories.
 """
 
-DEFAULT_ROOT_TESTSESSION_PATH = '/tmp/firecracker_test_session/'
+DEFAULT_ROOT_TESTSESSION_PATH = '/srv/tmp/firecracker_test_session/'
```
Comment (aghecenco) [inline tests/conftest.py]: Why does this have to change?

Comment (sandreim) [reply tests/conftest.py]: When running block tests the dev container needs to be started with -r and that will create a ramdisk where the block device file lives. 
So this change actually makes this possible to have the temporary block device filesystem stay in the ramdisk.

**Comment by acatangiu** [comment]
@dhrgit please take a look at this PR when you get the chance, and help with seeing if it's compatible with the new CI flow or what it needs to be compatible.

**Pull request closed without merging**